indexing-hadoop/src/main/java/org/apache/druid/indexer/HadoopTuningConfig.java

```diff
@@ -130,7 +142,8 @@ public HadoopTuningConfig(
       final @JsonProperty("logParseExceptions") @Nullable Boolean logParseExceptions,
       final @JsonProperty("maxParseExceptions") @Nullable Integer maxParseExceptions,
       final @JsonProperty("useYarnRMJobStatusFallback") @Nullable Boolean useYarnRMJobStatusFallback,
-      final @JsonProperty("awaitSegmentAvailabilityTimeoutMillis") @Nullable Long awaitSegmentAvailabilityTimeoutMillis
+      final @JsonProperty("awaitSegmentAvailabilityTimeoutMillis") @Nullable Long awaitSegmentAvailabilityTimeoutMillis,
+      final @JsonProperty("samplingFactor") int samplingFactor
```

Comment:
Maybe rename this config to `determinePartitionsSamplingFactor` to avoid ambiguity.

Reply:
Also, I think the property should be nullable, if not specified, it would default to 1.

Reply:
Done